op-chain-ops/genesis/config.go

```diff
@@ -390,13 +399,19 @@ func (d *DeployConfig) Check() error {
 		log.Warn("GasPriceOracleBlobBaseFeeScalar is 0")
 	}
 	if d.EIP1559Denominator == 0 {
-		return fmt.Errorf("%w: EIP1559Denominator cannot be 0", ErrInvalidDeployConfig)
+		log.Info("Using default EIP1559Denominator", "value", DefaultEIP1559Denominator)
+		d.EIP1559Denominator = DefaultEIP1559Denominator
 	}
-	if d.L2GenesisCanyonTimeOffset != nil && d.EIP1559DenominatorCanyon == 0 {
-		return fmt.Errorf("%w: EIP1559DenominatorCanyon cannot be 0 if Canyon is activated", ErrInvalidDeployConfig)
+	if d.EIP1559DenominatorCanyon == 0 {
+		log.Info("Using default EIP1559DenominatorCanyon", "value", DefaultEIP1559DenominatorCanyon)
+		d.EIP1559DenominatorCanyon = DefaultEIP1559DenominatorCanyon
 	}
 	if d.EIP1559Elasticity == 0 {
-		return fmt.Errorf("%w: EIP1559Elasticity cannot be 0", ErrInvalidDeployConfig)
+		log.Info("Using default EIP1559Elasticity", "value", DefaultEIP1559Elasticity)
+		d.EIP1559Elasticity = DefaultEIP1559Elasticity
+	}
+	if d.L2GenesisCanyonTimeOffset != nil && d.EIP1559DenominatorCanyon == 0 {
+		return fmt.Errorf("%w: EIP1559DenominatorCanyon cannot be 0 if Canyon is activated", ErrInvalidDeployConfig)
```

Comment:
Why not preserve the previous behaviour, where we only overwrite with the default if Canyon is activated? Seems a bit strange to _always_ override a zero value with the default (L407), and then later return an error if Canyon is activated and `d.EIP1559DenominatorCanyon == 0` (as written, I don't think we would ever hit this line, because `d.EIP1559DenominatorCanyon` will always be set to a nonzero value).